sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1852,6 +1852,24 @@ object SQLConf {
       .createWithDefault(
         "org.apache.spark.sql.execution.streaming.state.HDFSBackedStateStoreProvider")
 
+  val ENABLE_STATE_STORE_MAINTENANCE_THREAD_POOL =
+    buildConf("spark.sql.streaming.stateStore.enableStateStoreMaintenanceThreadPool")
+      .internal()
+      .doc("Whether background maintenance tasks are executed by single thread or thread pool")
```

Comment:
Shall we provide the context to inform the reason the old mechanism is set to default? Something along the line (please refine the sentence): the thread pool is consuming more resource than single thread, which might affect the performance of the query. Enable this if you observe the issue with the maintenance task.